browser_use/dom/dom_tree/index.js

```diff
@@ -845,23 +845,35 @@
       }
     }
 
-    // For elements in viewport, check if they're topmost
-    const centerX = rects[Math.floor(rects.length / 2)].left + rects[Math.floor(rects.length / 2)].width / 2;
-    const centerY = rects[Math.floor(rects.length / 2)].top + rects[Math.floor(rects.length / 2)].height / 2;
-
-    try {
-      const topEl = document.elementFromPoint(centerX, centerY);
-      if (!topEl) return false;
+    const margin = 5;
+    const rect = rects[Math.floor(rects.length / 2)];
+
+    // For elements in viewport, check if they're topmost. Do the check in the
+    // center of the element and at the corners to ensure we catch more cases.
+    const checkPoints = [
+      // Initially only this was used, but it was not enough
+      { x: rect.left + rect.width / 2, y: rect.top + rect.height / 2 },
+      { x: rect.left + margin, y: rect.top + margin },
+      { x: rect.right - margin, y: rect.top + margin },
+      { x: rect.left + margin, y: rect.bottom - margin },
+      { x: rect.right - margin, y: rect.bottom - margin }
```

Comment:
```suggestion
      { x: rect.left + margin, y: rect.top + margin },        // top left
      // { x: rect.right - margin, y: rect.top + margin },    // top right
      // { x: rect.left + margin, y: rect.bottom - margin },  // bottom left
      { x: rect.right - margin, y: rect.bottom - margin },    // bottom right
```

lets lower it to 2 additional click locations for now, I think 5 total is excessive as we need to check thousands of elements